src/pages/Dashboard/MintRepayVai/RepayVai/repayFee.tsx

```diff
@@ -0,0 +1,75 @@
+/** @jsxImportSource @emotion/react */
+import BigNumber from 'bignumber.js';
+import { LabeledInlineContent } from 'components';
+import React, { useCallback, useContext } from 'react';
+import { useTranslation } from 'translation';
+import { convertTokensToWei, convertWeiToTokens } from 'utilities';
+
+import { useGetVaiCalculateRepayAmount } from 'clients/api';
+import { TOKENS } from 'constants/tokens';
+import { AuthContext } from 'context/AuthContext';
+import useDebounceValue from 'hooks/useThrottle';
+
+import { useStyles } from '../styles';
+
+const DEBOUNCE_DELAY = 300;
+
+export interface IRepayFeeProps {
+  repayAmountTokens: string;
+}
+
+const RepayFee = ({ repayAmountTokens }: IRepayFeeProps) => {
+  const { t } = useTranslation();
+  const styles = useStyles();
+  const { account } = useContext(AuthContext);
+
+  const debouncedAmountTokens = useDebounceValue(repayAmountTokens, DEBOUNCE_DELAY);
+
+  const { data: vaiRepayAmountData } = useGetVaiCalculateRepayAmount(
+    {
+      accountAddress: account?.address || '',
+      repayAmountWei: convertTokensToWei({
+        value: new BigNumber(debouncedAmountTokens || 0),
+        token: TOKENS.vai,
+      }),
+    },
+    {
+      enabled: !!account?.address,
+      keepPreviousData: true,
+    },
+  );
+
+  const repayFeeWei = (vaiRepayAmountData?.vaiCurrentInterestWei || new BigNumber(0)).plus(
+    vaiRepayAmountData?.vaiPastInterestWei || 0,
+  );
+
+  const feePercentage = vaiRepayAmountData?.feePercentage || 0;
+
+  const getReadableRepayFee = useCallback(() => {
+    const fee = convertWeiToTokens({
+      valueWei: repayFeeWei,
+      token: TOKENS.vai,
+      minimizeDecimals: true,
+      returnInReadableFormat: true,
+    });
+
+    const formattedFeePercentage = `${new BigNumber(feePercentage)
+      .dp(feePercentage < 0.01 ? 6 : 2)
+      .toNumber()}%`;
+
+    return `${fee} (${formattedFeePercentage})`;
+  }, [repayFeeWei, feePercentage]);
```

Comment:
Ultimately all this logic depends on `vaiRepayAmountData?.vaiCurrentInterestWei`, so you can memomize it all based on it:

```tsx
const readableRepayFee = useMemo(() => {
  const repayFeeWei = new BigNumber(vaiRepayAmountData?.vaiCurrentInterestWei || 0).plus(
    vaiRepayAmountData?.vaiPastInterestWei || 0,
  );


  const fee = convertWeiToTokens({
    valueWei: repayFeeWei,
    token: TOKENS.vai,
    minimizeDecimals: true,
    returnInReadableFormat: true,
  });

  const feePercentage = vaiRepayAmountData?.feePercentage || 0;
  const formattedFeePercentage = `${new BigNumber(feePercentage)
    .dp(feePercentage < 0.01 ? 6 : 2)
    .toNumber()}%`;

  return `${fee} (${formattedFeePercentage})`;
}, [vaiRepayAmountData?.feePercentage]);
```